### DISCUSSION TITLE
Understanding how MQNs are calculated

### DISCUSSION THREAD

**Discussion by gianmarco-terrones** [rdkit/rdkit#6600] [Q&A] [answered]
I use RDKit to generate molecular quantum numbers (MQNs) and was hoping to see how it actually does the calculations. I see in RDKit documentation that MQNs are implemented in C++, and I also see that the file associated with `rdkit.Chem.rdMolDescriptors`'s function `MQNs_` is `rdkit/Chem/rdMolDescriptors.so`. However, I can't make much of this file. I am not too familiar with C++, so is there another way to better understand how MQNs like H-bond acceptor sites (MQN-20) are calculated?

**Comment by ptosco** [comment, marked as answer]
You may find the source code for `calcMQNs()` here:
https://github.com/rdkit/rdkit/blob/f5c9fb6a3144d37a0487b78781995bd8d48da795/Code/GraphMol/Descriptors/MQN.cpp#L20
